libs/github/go.mod

```diff
@@ -0,0 +1,32 @@
+module github.com/gravitational/shared-workflows/libs/github
+
+go 1.22.4
+
+require (
+	github.com/cli/go-gh/v2 v2.9.0
+	github.com/google/go-github/v37 v37.0.0
+	github.com/gravitational/trace v1.3.1
+	github.com/shurcooL/githubv4 v0.0.0-20240429030203-be2daab69064
```

Comment:
Why do we need 3 different libraries for Github? Can we simplify? In our existing bot we already use github.com/google/go-github.

Reply:
I was using the v4 API for the code but it isn't absolutely necessay so just moved it real quick to use the v3 API. The `go-gh` library has a useful function I just stumbled upon that will use look for `GITHUB_TOKEN`, github config file, or `gh auth`. Useful for running it on developer machines and CI.

I could remove that dependency. I need to update the config to note that `GITHUB_TOKEN` is needed anyway. Could have it maybe be a requirement to pass in the token.